src/core/thread/indirect_sender.cpp

```diff
@@ -582,34 +697,27 @@ Error IndirectSender::PrepareFrameForCslNeighbor(Mac::TxFrame &aFrame,
     Error error = kErrorNotFound;
 
 #if OPENTHREAD_FTD
-    // `CslNeighbor` can only be a `Child` for now, but can be changed later.
-    error = PrepareFrameForChild(aFrame, aContext, static_cast<Child &>(aCslNeighbor));
+    if (IsChild(aCslNeighbor))
+    {
+        error = PrepareFrameForChild(aFrame, aContext, static_cast<Child &>(aCslNeighbor));
+        VerifyOrExit(error != kErrorNone);
```

Comment:
```suggestion
        ExitNow();
```
We don't want to then call `PrepareFrameForEnhCslNeighbor` even if `PrepareFrameForChild` fails, right (for FTD + WED case)

Reply:
That's right, at this point we already know that we are dealing with a child.